app/src/main/java/org/sagebionetworks/research/mpower/authentication/ExternalIdSignInActivity.java

```diff
@@ -0,0 +1,102 @@
+package org.sagebionetworks.research.mpower.authentication;
+
+import static android.support.design.widget.Snackbar.LENGTH_SHORT;
+
+import android.arch.lifecycle.ViewModelProviders;
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.v7.app.AppCompatActivity;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.ProgressBar;
+
+import com.google.common.base.Strings;
+
+import org.sagebionetworks.research.mpower.R;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnCheckedChanged;
+import butterknife.OnClick;
+import butterknife.OnTextChanged;
+import dagger.android.AndroidInjection;
+
+public class ExternalIdSignInActivity extends AppCompatActivity {
+    ExternalIdSignInViewModel externalIdSignInViewModel;
+
+    @Inject
+    ExternalIdSignInViewModel.Factory externalIdSignInViewModelFactory;
+
+    @BindView(R.id.progressBar)
+    ProgressBar progressBar;
+
+    @BindView(R.id.signIn)
+    Button signIn;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        AndroidInjection.inject(this);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_external_id_sign_in);
+        ButterKnife.bind(this);
+
+        externalIdSignInViewModel = ViewModelProviders.of(this, externalIdSignInViewModelFactory.create())
+                .get(ExternalIdSignInViewModel.class);
+
+        externalIdSignInViewModel.getIsSignedInLiveData().observe(this, isSignedIn -> {
+            if (isSignedIn) {
+                finish();
+            }
+        });
+
+        externalIdSignInViewModel.getIsLoadingLiveData().observe(this, isLoading -> {

```

Comment:
Do you always have to check the state the live data when you create your activity?  I thought that the ViewModel sends out its last state when you observe it.

Reply:
Oh, it looks like I was mistaken with what was going on here.